### PR TITLE
Fix issues of weight-decay

### PR DESCRIPTION
**Pull request by cheungdaven**
1. In MXNet, the concise implementation of using weight decay generates a plot that looks like no weight decay is used.
weight decay parameter is added.

2. In TensorFlow, the concise implementation and from-scratch implementation plots (using weight decay) do not align with each other well.
The hyper-parameter is tuned.

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice.

### CODE REVIEW

File: chapter_linear-networks/weight-decay.md [hunk 1]
```diff
@@ -412,14 +413,16 @@ and this work becomes more routine.
 
 ```{.python .input  n=14}
 %%tab all
-if tab.selected('mxnet') or tab.selected('tensorflow'):    
+if tab.selected('mxnet'):
     model = WeightDecay(wd=3, lr=0.01)
+if tab.selected('tensorflow'):
+    model = WeightDecay(wd=0.3, lr=0.01)
```
Comment (astonzhang) [inline chapter_linear-networks/weight-decay.md:419]: Is it possible to keep the wd the same for all the frameworks?

File: chapter_linear-networks/weight-decay.md [hunk 2]
```diff
@@ -369,11 +369,12 @@ class WeightDecay(d2l.LinearRegression):
     def __init__(self, wd, lr):
         super().__init__(lr)
         self.save_hyperparameters()
+        self.wd = wd
         
     def configure_optimizers(self):
         self.collect_params('.*bias').setattr('wd_mult', 0)
         return gluon.Trainer(self.collect_params(),
-                             'sgd', {'learning_rate': self.lr})
+                             'sgd', {'learning_rate': self.lr,'wd': self.wd})
```
Comment (astonzhang) [inline chapter_linear-networks/weight-decay.md:377]: Add a space after the comma

**Comment by astonzhang** [comment]
Thanks. Do we also need to improve the PyTorch code so that the plots of using wd for both from-scratch & concise implementations are more consistent with each other?

File: chapter_linear-networks/weight-decay.md (hunk 1 above)
Comment (cheungdaven) [reply chapter_linear-networks/weight-decay.md:419]: For tensorflow, the wd is a L2 regularization rate. For pytorch and mxnet, the regularization rate is defined as wd \times lr. So, actually, setting wd of tensorflow to 0.03 is equivalent to setting it to 3 in pytorch and mxnet. But I found that setting it to 0.3 is more visible in the plot so I chose 0.3 instead of 0.03.

Comment (astonzhang) [reply chapter_linear-networks/weight-decay.md:419]: "setting wd of tensorflow to 0.03 is equivalent to setting it to 3 in pytorch and mxnet"
Can you provide more context? Can you also add the explanations in the main text right above the code?

**Comment by astonzhang** [comment]
Once https://github.com/d2l-ai/d2l-en/pull/2035/files is merged to master, please rebase this PR to master to include those changes (in case you wish to refer to them when explaining how the "shrinkage" works when the wd hyperparam is specified).

File: chapter_linear-networks/weight-decay.md (hunk 1 above)
Comment (cheungdaven) [reply chapter_linear-networks/weight-decay.md]: I double checked the tensorflow implementation and I found that it is also l2 regularization so I changed the wd back to 3. It is the weight initialization that effects the plot so I added the normal initialization.

File: chapter_linear-networks/weight-decay.md [hunk 3]
```diff
@@ -396,7 +398,9 @@ class WeightDecay(d2l.LinearRegression):
         super().__init__(lr)
         self.save_hyperparameters()
         self.net = tf.keras.layers.Dense(
-            1, kernel_regularizer=tf.keras.regularizers.l2(wd))
+            1, kernel_regularizer=tf.keras.regularizers.l2(wd),
+            kernel_initializer=tf.keras.initializers.RandomNormal(0.0, 0.01)
```
Comment (astonzhang) [inline chapter_linear-networks/weight-decay.md:402]: 0.0 -> 0 for simplicity if possible

Comment (cheungdaven) [reply chapter_linear-networks/weight-decay.md]: resolved

**Review by AnirudhDagar — APPROVED** [review]
Plots are consistent. LGTM! Thanks @cheungdaven

**Comment by AnirudhDagar** [comment]
@d2l-bot please rebuild!
